src/Charcoal/Admin/Widget/QuickFormWidget.php

```diff
@@ -50,6 +85,146 @@ public function submitLabel()
         return parent::submitLabel();
     }
 
+    /**
+     * Determine if the form has any multilingual properties.
+     *
+     * @return boolean
+     */
+    public function hasL10nFormProperties()
+    {
+        if ($this->validateObjType()) {
+            $locales = count($this->translator()->availableLocales());
+            if ($locales > 1) {
+                $obj = $this->obj();
+
+                $props = [];
+
+                foreach ($this->groups as $group) {
+                    array_push($props, ...($group->groupProperties()));
+                }
+
+                $props = array_values(array_unique($props));
+
+                foreach ($props as $prop) {
+                    $modelProp = $obj->property($prop);
+
+                    if ($modelProp['l10n']) {
+                        return true;
+                    } elseif ($modelProp instanceof ModelStructureProperty) {
+                        $metadata = $modelProp->getStructureMetadata();
+                        foreach ($metadata->properties() as $prop) {
+                            if (isset($prop['l10n']) && $prop['l10n']) {
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        return parent::hasL10nFormProperties();
+    }
+
+    /**
+     * @return boolean
+     */
+    public function showLanguageSwitch()
+    {
+        if ($this->showLanguageSwitch === null) {
+            $this->showLanguageSwitch = $this->hasL10nFormProperties();
+        }
+        return $this->showLanguageSwitch;
+    }
+
+    /**
+     * Determine if content groups are to be displayed as languages tabbable panes.
+     *
+     * @return boolean
+     */
+    public function isDisplayModeLang()
+    {
+        return ($this->groupDisplayMode() === self::DISPLAY_MODE_LANG) ? true : false;
```

Comment:
In PHP, a comparison expression results in a `boolean`, no need for a ternary expression.

```suggestion
        return ($this->groupDisplayMode() === self::DISPLAY_MODE_LANG);
```